lib/SILOptimizer/Mandatory/TFConstExpr.cpp

```diff
@@ -814,19 +814,28 @@ ConstExprFunctionState::computeCallResult(ApplyInst *apply) {
       elementConstants.assign(numElements, SymbolicValue::getUninitMemory());
     } else {
       // We handle the flow-insensitive case specially in order to find the
-      // stores to initialize the array elements.  Collect them here and pretend
-      // that the array was initialized atomically.
-      SmallVector<SILValue, 8> elements;
+      // stores/apply's to initialize the array elements.  Collect them here
+      // and pretend that the array was initialized atomically.
+      SmallVector<Operand*, 8> elementsAtInit;
       if (ConstExprEvaluator::decodeAllocUninitializedArray(
-              apply, numElements, elements,
+              apply, numElements, elementsAtInit,
               /*arrayInsts*/ nullptr))
         return evaluator.getUnknown((SILInstruction *)apply,
                                     UnknownReason::Default);
 
       // Okay, we were able to decode the array.  See if we can fold all of the
       // elements we found.
-      for (auto elt : elements) {
-        auto eltCst = getConstantValue(elt);
+      for (auto *use : elementsAtInit) {
+        SymbolicValue eltCst = evaluator.getUnknown((SILInstruction *)apply,
+                                                    UnknownReason::Default);
+        if (auto *store = dyn_cast<StoreInst>(use->getUser())) {
+          eltCst = getConstantValue(store->getSrc());
+        } else {
+          auto addr = computeSingleStoreAddressValue(use->get());
+          if (addr.getKind() == SymbolicValue::Address)
+            eltCst = addr.getAddressValueMemoryObject()->getValue();
```

Comment:
Yes, it works! `computeLoadResult` is actually much better here. Because `computeSingleStoreAddressValue` is supposed to accept `alloc`s only; if we want to pass a non-`alloc` address to it, we'd need to modify its implementation to break such restriction.